crates/loader/src/local/assets.rs

```diff
@@ -20,22 +20,49 @@ use super::config::{RawDirectoryPlacement, RawFileMount};
 pub(crate) async fn prepare_component(
     raw_mounts: &[RawFileMount],
     src: impl AsRef<Path>,
-    base_dst: impl AsRef<Path>,
+    base_dst: Option<impl AsRef<Path>>,
     id: &str,
     exclude_files: &[String],
 ) -> Result<Vec<DirectoryMount>> {
-    tracing::info!(
-        "Mounting files from '{}' to '{}'",
-        src.as_ref().display(),
-        base_dst.as_ref().display()
-    );
-
-    let files = collect(raw_mounts, exclude_files, src)?;
-    let host = create_dir(&base_dst, id).await?;
-    let guest = "/".to_string();
-    copy_all(&files, &host).await?;
+    if let Some(base_dst) = base_dst {
+        tracing::info!(
+            "Mounting files from '{}' to '{}'",
+            src.as_ref().display(),
+            base_dst.as_ref().display()
+        );
 
-    Ok(vec![DirectoryMount { guest, host }])
+        let files = collect(raw_mounts, exclude_files, src)?;
+        let host = create_dir(&base_dst, id).await?;
+        let guest = "/".to_string();
+        copy_all(&files, &host).await?;
+
+        Ok(vec![DirectoryMount { guest, host }])
+    } else {
+        tracing::info!("directly mounting local asset directories into guest");
+
+        if !exclude_files.is_empty() {
+            bail!("exclusions not permitted when mounting directories directly")
```

Comment:
This line means that if you ask for direct mounts for an app that specifies on or more files to be excluded, we'll raise an error and exit without running the app.

Per [Ivan's comment](https://github.com/fermyon/spin/issues/964#issuecomment-1347186000), we'll need a more general "hot reload" solution to handle exclusions correctly.  It will probably involve using a temporary directory and simultaneously monitoring the original directories and files, copying the non-excluded items over to the temporary directory as they change.